src/mame/ibm/ibm3477.cpp

```diff
@@ -0,0 +1,118 @@
+   ___| |__|  |     |  |     |                     |___________|             |
+ _|__         |_____|  |_____|                                   ____        |
+|   |                                                      10124N-> |        |
+|___|         ||||||||||||||||                                   |  |        |
+  |                                                              |__|        |
+  |                            _________________    ___               ___    |
+  |___________________________|                |_|_|  |               |  |___|
+                                                      |_|_|_|_|_|_|_|_|
+
+ToDo:
+- Everything!
+
+********************************************************************************/
+
+#include "emu.h"
+#include "cpu/i86/i86.h"
+#include "emupal.h"
+#include "screen.h"
+
+
+namespace {
+
+class ibm3477_state : public driver_device
+{
+public:
+	ibm3477_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		, m_maincpu(*this, "maincpu")
+		, m_p_chargen(*this, "chargen")
+		{ }
+
+	void ibm3477(machine_config &config);
+
+private:
+	virtual void machine_reset() override ATTR_COLD;
+	void ibm3477_palette(palette_device &palette) const;
+	uint32_t screen_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect);
+
+	required_device<cpu_device> m_maincpu;
+	required_region_ptr<u8> m_p_chargen;
+};
+
+
+// Input ports
+static INPUT_PORTS_START( ibm3477 )
+INPUT_PORTS_END
+
+uint32_t ibm3477_state::screen_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect)
+{
+	return 0;
+}
+
+void ibm3477_state::ibm3477_palette(palette_device &palette) const
+{
+	palette.set_pen_color(0, 0,   0, 0); // Black
+	palette.set_pen_color(1, 0, 255, 0); // Full
+	palette.set_pen_color(2, 0, 128, 0); // Dimmed
+}
+
+void ibm3477_state::machine_reset()
+{
+}
+
+void ibm3477_state::ibm3477(machine_config &config)
+{
+	// Basic machine hardware
+	I8088(config, m_maincpu, 59'818'500 / 10); // Unknown clock
+
+	// Video hardware
+	screen_device &screen(SCREEN(config, "screen", SCREEN_TYPE_RASTER));
+	screen.set_refresh_hz(60);
+	screen.set_vblank_time(ATTOSECONDS_IN_USEC(2500)); // Not accurate
+	screen.set_screen_update(FUNC(ibm3477_state::screen_update));
+	screen.set_size(640, 240);
+	screen.set_visarea(0, 639, 0, 239);
+	screen.set_palette("palette");
+
+	PALETTE(config, "palette", FUNC(ibm3477_state::ibm3477_palette), 3);
+}
+
+// ROM definition
+ROM_START( ibm3477 )
+	ROM_REGION( 0x40000, "maincpu", 0 )
+	ROM_LOAD("pn_38f5843_ibm_30-10-89.zm12", 0x00000, 0x20000, CRC(b07efb93) SHA1(84d245d17828bc4e096e0cae2d066f2209f56a8f) )
+	ROM_LOAD("pn_38f5844_ibm_20-10-89.zm13", 0x20000, 0x20000, CRC(05f3c0f6) SHA1(f99096a2f3e97fe1eb540db3f3e620ddccf71bbc) )
+
+	ROM_REGION( 0x2000, "chargen", 0 )
+	ROM_LOAD( "char.bin", 0x0000, 0x2000, NO_DUMP ) // probably inside the video processor
```

Comment:
Has an attempt been made for the roms to checkout if they `gfxdecode` cleanly with a 1bpp linear layout?

Reply:
No

Reply:
Before your `machine_config`:
```c++
static GFXDECODE_START( gfx_ibm3477 )
	GFXDECODE_ENTRY("maincpu", 0x0000, gfx_8x8x1, 0, 1)
GFXDECODE_END
```
Then, in your machine_config, around palette declaration:
```c++
	GFXDECODE(config, "gfxdecode", "palette", gfx_ibm3477);
```
F4 menu, press enter once, scroll down with page up/down and check if there's some kind of clear GFX in there. If it does then the HW has some kind of Programmable Character Generator so the `NO_DUMP` is redundant.

Reply:
Thanks!!

I see some loose characters, but cannot find a complete charset... Enough to say that there's not a missing separate character ROM?

![image](https://github.com/user-attachments/assets/dc708ccd-d237-4bd3-8968-676a790d1b47)
![image](https://github.com/user-attachments/assets/ca9987ee-735b-437e-9383-d6f23c79827c)
![image](https://github.com/user-attachments/assets/438d1059-7610-42ee-bcdd-67cbf38aa375)